src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/BedSearchService.kt

```diff
@@ -143,13 +143,35 @@ class BedSearchService(
           when (it) {
             BedSearchAttributes.singleOccupancy -> "Single occupancy"
             BedSearchAttributes.sharedProperty -> "Shared property"
-            BedSearchAttributes.wheelchairAccessible -> "Wheelchair accessible"
+            BedSearchAttributes.wheelchairAccessible -> ""
           }
         }
 
+        val roomCharacteristicsNames = when {
+          propertyBedAttributes?.contains(BedSearchAttributes.wheelchairAccessible) == true -> listOf("Wheelchair accessible")
+          else -> null
+        }
+
         val premisesCharacteristicIds = premisesCharacteristicsNames?.let {
           val premisesCharacteristics = characteristicService.getCharacteristicsByNames(premisesCharacteristicsNames)
-          premisesCharacteristics.filter { it.isActive && it.matches(ServiceName.temporaryAccommodation.value, "premises") }.map { it.id }.toList()
+          premisesCharacteristics.filter {
+            it.isActive &&
+              (
+                it.matches(ServiceName.temporaryAccommodation.value, "premises") ||
+                  it.matches(ServiceName.temporaryAccommodation.value, "*")
+                )
+          }.map { it.id }.toList()
+        } ?: emptyList()
+
+        val roomCharacteristicIds = roomCharacteristicsNames?.let {
+          val roomCharacteristics = characteristicService.getCharacteristicsByNames(roomCharacteristicsNames)
+          roomCharacteristics.filter {
+            it.isActive &&
+              (
+                it.matches(ServiceName.temporaryAccommodation.value, "room") ||
+                  it.matches(ServiceName.temporaryAccommodation.value, "*")
+                )
+          }.map { it.id }.toList()
```

Comment:
these two chunks of code to resolve characteristics ids appear to be almost identical other than matching on 'room' instead of 'premises'. could this be moved into a common private function?